### PR TITLE
CI-220: Results page

### PR DESCRIPTION
**Pull request by SerhiiBorozenets**
[CI-220](https://cloverpop-internship.atlassian.net/browse/CI-220)

[CI-220]: https://cloverpop-internship.atlassian.net/browse/CI-220?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

NOTE: change rake task name in scheduler from `send_results_email:send_results_email` to `notifications:send_results_email`

### CODE REVIEW

File: app/controllers/api/v1/results_controller.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    @time_period ||= TimePeriod.find_by(id: params[:id])
+  end
+
+  def user
+    @user ||= User.find_by(id: params[:user_id])
+  end
+
+  def sign_in_user
+    sign_in user
+  end
```
Comment (rogergraves) [inline app/controllers/api/v1/results_controller.rb:30]: I'm not sure what the advantage is of having a method for this specifically, since `sign_in user` looks the same as `sign_in_user`.

File: app/assets/stylesheets/gif.scss [hunk 2]
```diff
@@ -115,6 +121,14 @@
   }
 }
 
+.gif-result-card {
+  border: none !important;
```
Comment (SokalMet) [inline app/assets/stylesheets/gif.scss:125]: this style could be pretty common, so I recommend to create 'border-none' class and use it on place of need.

File: app/assets/stylesheets/gif.scss [hunk 3]
```diff
@@ -115,6 +121,14 @@
...
+  .card-full {
+    height:  auto;
+    padding: 0;
+  }
```
Comment (SokalMet) [inline app/assets/stylesheets/gif.scss:129]: you can use 'h-auto' and 'p-0' class names instead   .card-full

**Review by SokalMet — APPROVED** [review]

File: app/controllers/api/v1/results_controller.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+  def current_response
+    @current_response ||= Response.find_by(time_period_id: TimePeriod.current.id, user_id: current_user.id)
```
Comment (SBorozenets) [inline app/controllers/api/v1/results_controller.rb:33]: try to rewrite `current_user.responses.find_by(time_period_id: TimePeriod.current.id)`

File: app/models/time_period.rb [hunk 5]
```diff
@@ -37,6 +40,11 @@ def current
     def find_or_create_time_period
       TimePeriod.current || TimePeriod.create_time_period
     end
+
+    def previous_time_period
+      find_or_create_time_period
+      TimePeriod.find_by(end_date: TimePeriod.current.start_date - 1)
```
Comment (SBorozenets) [inline app/models/time_period.rb:46]: maybe it's better `TimePeriod.current.end_date.week_ago`

File: spec/requests/results_spec.rb [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    passwordless_sign_in(user)
+  end
+
+  describe 'GET #show' do
+    subject { get "/api/v1/results/#{time_period.id}" }
+
+    it 'returns a success response' do
+      subject
+      expect(response).to have_http_status(:success)
+    end
```
Comment (SBorozenets) [inline spec/requests/results_spec.rb:18]: please add test coverage on negative case

**Review by SBorozenets — APPROVED** [review]

**Comment by SerhiiBorozenets** [comment]
codeclimate issues are not relevant
